docs/CustomWidget.md

```diff
@@ -0,0 +1,86 @@
+
+```typescript
+type Style = {
+    // color palette
+    palette: { 
+        accent: string;
+        primary: string;
+        secondary: string;
+        text: string;
+        textAccent: string 
+    };
+    // fonts, custom fonts may be used
+    fontPalette: { 
+        primary: string;
+        secondary: string;
+    };
+};
+```
+
+### Accessing Constructor Arguments
+
+```Javascript
+export default class FeaturedSessionWidget extends HTMLElement {
+    constructor({ configuration, theme }) {
+        super();
+        // store a custom field from our configuration as a class variable
+        this.featuredSessionIds = this.configuration.featuredSessionIds ?? [];
+
+        // store the entire theme for use elsewhere in our custom element
+        this.theme = theme;
+    }
+}
+```
+
+## Accessing the Custom Widgets SDK <a name="accessSDK"></a>
+
+Behind the scenes, methods in the SDK are provided as mixins to the CE class. This means that the methods defined in the [custom widgets SDK](./CustomWidgetsSDK.md) will also be available as methods on the CE class.
+
+Since the CE constructor cannot await asynchronous function calls, you'll probably want to call these methods elsewhere. One way to do this is by using [HTML custom element lifecycle callbacks](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements#using_the_lifecycle_callbacks). Using the `connectedCallback` lifecycle method, we can call and await an async SDK method when our custom element is added to the document:
+
+```javascript
+  async connectedCallback() {
+    const sessionDetails = await this.getSessionDetails();
+  }
+```
+
+## Defining Multiple Custom Elements
+
+You may want to use multiple custom elements to better organize your code for reuse. You can import other CE classes into the file where your entrypoint CE is defined. Be sure to include the `.js` file extension in the import path.
+
+Accessory CEs must be defined in the document's custom element registry using a valid name that does not conflict with the name of a custom element for _any_ custom widget. Always check whether the accessory CE has already been defined by another instance of the widget element, attempting to define an element twice will result in an error.
+
+## Using Shadow DOM
+
+The Web Components standard includes the shadow dom feature which can be used to encapsulate the scripting and styling of the custom element to prevent interference with other aspects fo the registration site.
+
+```Javascript
+export default class FeaturedSessionWidget extends HTMLElement {
+    constructor({ configuration, theme }) {
+        super();
+        // Create a shadow root
+        this.attachShadow({ mode: "open" });
+
+        const hello = document.createElement('h1');
+        hello.textContent = "Hello World"
```

Comment:
Should we consume some data or something here?

Reply:
I originally intended this as just an example of shadow dom usage, but a more fleshed out example of a custom widget element implementation could include this and more. going to update